Added duplicated GeneFamily member information to LevelComparisonResult

qfma opened the pull request:
Dear all,

I have added functionality to the LevelComparisonResult Class to print the members of a duplicated GeneFamily, and the members of the new sub-families. Previously it would only show that a duplication event occurred at a given node, but not the content of these families.

Sample output for a duplicated family, when printing the results of
a LevelComparison looks like this:

```
....
112: lost
113: identical
-------
114 --> 114.2a; 114.2b
114: 47953; 28082; 11418; 29862; 117; 50097; 13634; 50845; 14300
114.2a: 50097; 13634
114.2b: 50845; 14300
-------
...
```

The changes also enhance the FamDupl object, which now uses GeneFamily objects, rather than the LOFT annotation. This allows the retention of the GeneFamily information, which can be easily accessed using the LevelComparisonResult.filter() function. Additional modifications would be needed for all other FamEvent children (eg. FamLost, FamIdent etc.) If this is something that would be useful, I would be happy to work on this.

If anything is unclear, or futher modifications are requested, just let me know!

Best,
Fabian

qfma committed:
Added functionality to the LevelComparisonResult Class to print the
members of a duplicated GeneFamily, and the members of the new sub-
families.

Sample output for a duplicated family, when printing the results of
a LevelComparison looks like this:

Sample output:
-------
114 --> 114.2a; 114.2b
114: 47953; 28082; 11418; 29862; 117; 50097; 13634; 50845; 14300
114.2a: 50097; 13634
114.2b: 50845; 14300
-------

alpae commented:
Hi Fabian,
thanks for the contribution. The changes you introduced make some of the existing tests fail. nothing serious, but it has to do with the new way you initialize the FamDupl object afaiks.
If you please could update the failing tests and maybe add one that tests about the duplicated members. Thanks
Adrian

qfma commented:
Of course, I completely forgot about that. Will get back to you once I updated the tests.

qfma committed:
Changed FamEvent initialization
Added a str representation function to the GeneFamily object.
Removed explicit checking of object types via isinstance()
Created .name field for FamEvent object.

qfma committed:
Fixed tests
Replaced .fam usage with .name. The .fam field contains the GeneFamily
objects and .name the FamID

qfma commented:
Hi Adrian,

I changed how the FamEvent/FamDupl object is initialized and fixed the tests.
Tests all pass when running:
'''
python -m unittest discover test/ 'test_*.py
'''
 This introduces a new field for the FamEvent object .name, which contains the FamIDs. The .fam field now holds the GeneFamily objects. Is that OK or can you think of a better way?

I will write some tests later this week to test the new code.

Cheers,
Fabian
